lib/apiservers/engine/backends/image.go

```diff
@@ -147,8 +155,22 @@ func (i *Image) PullImage(ctx context.Context, ref reference.Named, metaHeaders
 		return err
 	}
 
-	client := PortLayerClient()
-	ImageCache().Update(client)
+	imageJSON, err := ioutil.ReadFile(imageFile)
```

Comment:
Looks like instead of the temp file, you could use io.MultiWriter:
```go
var buf bytes.Buffer
cmd.Stdout = io.MultiWriter(outStream, &buf)
...
json.Unmarshal(buf.Bytes(), image)
```